arrow-flight/tests/flight_sql_client.rs

```diff
@@ -94,21 +92,26 @@ pub async fn test_execute_ingest() {
         make_primitive_batch(2),
     ];
     let actual_rows = flight_sql_client
-        .execute_ingest(cmd, batches)
+        .execute_ingest(cmd, futures::stream::iter(batches.clone()).map(Ok))
         .await
         .expect("ingest should succeed");
     assert_eq!(actual_rows, expected_rows);
+    // make sure the batches made it through to the server
+    let ingested_batches = test_server.ingested_batches.lock().await.clone();
+    assert_eq!(ingested_batches, batches);
 }
 
 #[derive(Clone)]
 pub struct FlightSqlServiceImpl {
     transactions: Arc<Mutex<HashMap<String, ()>>>,
+    ingested_batches: Arc<Mutex<Vec<RecordBatch>>>,
```

Comment:
I also propose extending the test slightly to ensure that the batches came through unchanged (in addition to the row count being sent back correctly)